core/src/main/java/org/jdbi/v3/InferredColumnMapperFactory.java

```diff
@@ -13,42 +13,36 @@
  */
 package org.jdbi.v3;
 
-import java.util.List;
+import static org.jdbi.v3.Types.findGenericParameter;
 
-import com.fasterxml.classmate.ResolvedType;
-import com.fasterxml.classmate.TypeResolver;
+import java.lang.reflect.Type;
 
 import org.jdbi.v3.tweak.ResultColumnMapper;
 
 class InferredColumnMapperFactory<X> implements ResultColumnMapperFactory
 {
-    private static final TypeResolver tr = new TypeResolver();
-    private final Class<X> maps;
+    private final Type maps;
     private final ResultColumnMapper<X> mapper;
 
-    @SuppressWarnings("unchecked")
     InferredColumnMapperFactory(ResultColumnMapper<X> mapper)
     {
+        this.maps = findGenericParameter(mapper.getClass(), ResultColumnMapper.class)
+                .orElseThrow(() -> new UnsupportedOperationException("Must use a concretely typed ResultColumnMapper here"));
         this.mapper = mapper;
-        ResolvedType rt = tr.resolve(mapper.getClass());
-        List<ResolvedType> rs = rt.typeParametersFor(ResultColumnMapper.class);
-        if (rs.isEmpty() || rs.get(0).getErasedType().equals(Object.class)) {
-            throw new UnsupportedOperationException("Must use a concretely typed ResultColumnMapper here");
-        }
-
-        maps = (Class<X>) rs.get(0).getErasedType();
     }
 
     @Override
-    public boolean accepts(Class<?> type, StatementContext ctx)
+    public boolean accepts(Type type, StatementContext ctx)
     {
         return maps.equals(type);
     }
 
-    @SuppressWarnings("unchecked")
     @Override
-    public <T> ResultColumnMapper<? extends T> columnMapperFor(Class<T> type, StatementContext ctx)
+    public ResultColumnMapper<?> columnMapperFor(Type type, StatementContext ctx)
     {
-        return (ResultColumnMapper<? extends T>) mapper;
+        if (!type.equals(maps)) {
```

Comment:
This was in `InferredResultSetMapperFactory`, I thought it was a good idea to copy it over.

I'm not sure how long it will be relevant though. Now that most factories have lost generic parameters, one of my next goals is to collapse the accept/build methods into a single method that possibly returns an Optional. At that point this check will be redundant anyway.